src/pages/LoginPage/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import { useAuth } from '../../AuthContextProvider';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faGoogle } from "@fortawesome/free-brands-svg-icons";
+
+interface LoginPageProps {
+  onSignUpClick: () => void;
+  onLoginSuccess: () => void;
+}
+
+export const LoginPage: React.FC<LoginPageProps> = ({ onSignUpClick, onLoginSuccess }) => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const { login, signInWithGoogle } = useAuth();
+
+  const togglePasswordVisibility = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
```

Comment:
I cannot see any validation for username and password inputs. Anything that a user may enter is just sent to firebase. It would be better to add some client side validation to guide the user on the username & password length and supported characters, for example.